jenkins/spark-tests.sh

```diff
@@ -376,6 +376,12 @@ if [[ $TEST_MODE == "DEFAULT" ]]; then
   EXPLAIN_ONLY_CPU_SMOKE_TEST=1 \
     ./run_pyspark_from_build.sh
 
+  # Spark Connect smoke test (available in Spark 3.5.6+)
+  if printf '%s\n' "3.5.6" "$SPARK_VER" | sort -V | head -1 | grep -q "3.5.6"; then
```

Comment:
nit: could leave the version check in run_pyspark script only to save maintenance efforts in the future